src/utils/api.js

```diff
@@ -0,0 +1,67 @@
+/* The API communication module.
+   Inspired by Svelte RealWorld:
+     https://github.com/sveltejs/realworld/blob/master/src/node_modules/api.js */
+
+export const API_HOST = process.env.SAPPER_APP_API_HOST;
+
+
+function request(method, url, data) {
+  const fetch = process.browser ? window.fetch : require('node-fetch').default;
+  const formDataClass = process.browser ? FormData : require('url').URLSearchParams;
+
+  let options = {
+    credentials: 'include',
+    method,
+    headers: {},
+  };
+
+  if (data instanceof formDataClass) {
+    options.body = data;
+  } else if (data != null) {
+    options.headers['Content-Type'] = 'application/json';
+    options.body = JSON.stringify(data);
+  }
+
+  return fetch(url, options)
+		.then(r => r.text())
+		.then(json => {
+			try {
+				return JSON.parse(json);
+			} catch (err) {
+				return json;
+			}
+		});
```

Comment:
Why not simply `r => r.json()`? I don't think there is any endpoint in the backend that doesn't return JSON (or no payload). And even if, better check the response headers for Content-Type

Reply:
This code was copy-pasted. And it seems alright to me, inspecting the Content-Type in the second callback would be impossible and putting it in the first seems kinda meh.

Reply:
What isn't alright is that it will return either JSON or just a string (probably HTML, if something is not properly handled in backend), when it should just return JSON. If it didn't, it should cause an error so that we find it and fix it rather than having something just not working properly and we spend hours trying to find the root cause